### DISCUSSION TITLE
Table rendering very slow

### DISCUSSION THREAD

**Discussion by MaximeGratens** [filamentphp/filament#6990] [Help]
Hey!

I have an issue that I can't fix, in prod and in local, rendering a table with 50 models is very slow.

It's not about the query, only 3 query and it's rendering in ms, it's about the rendering of the table it self.

I can see on sentry

view.render | tables::index | 100% | 2.11s | 2.11s
-- | -- | -- | -- | --

If someone have an idea about this.
(I already read every discussions about "slow filament"

**Comment by pboivin** [comment]
Can you pinpoint what is causing the slowdown? Is it one specific column?

**Comment by pboivin** [comment]
An idea: you can defer loading the table until the rest of the page is ready:
```php
->deferLoading()
```
Not a fix but maybe this can help the perceived performance of your page, if there's no other optimization?

**Comment by MaximeGratens** [comment]
re, sorry for the late answer,
I'm already using the deferLoading, but as I told you, the problem is not from our database (-1ms) but. from filament and livewire being very slow to render 

view.render | tables::index | 100% | 2.11s | 2.11s
-- | -- | -- | -- | --

**Reply by pboivin** [reply]
Got it. I'm wondering if you can maybe disable one column at a time in the Filament table, just to see if you can zoom in on what is causing the slowdown.

**Reply by danharrin** [reply]
Do you have Debugbar installed?

**Reply by joshbonnick** [reply]
@danharrin bit late to the discussion but that was my issue with performance, specifically client side performance rendering everything. 

Disabling debugbar for the filament routes completely resolved the issue.

**Comment by mark-nicepants** [comment]
I have the same issue

Sentry is tracing and pointing to view rendering aswell. I did ran laravel and filament optimize

![image](https://github.com/user-attachments/assets/cd65214f-607d-4880-a026-7c38958c16fc)

**Reply by mark-nicepants** [reply]
All subsequent traces are 480 small 1 a 2ms view renders.

**Reply by miagg** [reply]
I think I have the same [issue](https://github.com/filamentphp/filament/issues/9304#issuecomment-2419246646)
